prow/jira/jira.go

```diff
@@ -782,3 +785,19 @@ func (l *retryableHTTPLogrusWrapper) Debug(msg string, context ...interface{}) {
 func (l *retryableHTTPLogrusWrapper) Warn(msg string, context ...interface{}) {
 	l.log.WithFields(l.fieldsForContext(context...)).Warn(msg)
 }
+
+// Search wraps SearchWithContext using the background context.
+func (jc *client) Search(jql string, options *jira.SearchOptions) ([]jira.Issue, *jira.Response, error) {
+	return jc.SearchWithContext(context.Background(), jql, options)
+}
+
+func (jc *client) SearchWithContext(ctx context.Context, jql string, options *jira.SearchOptions) ([]jira.Issue, *jira.Response, error) {
+	issue, response, err := jc.upstream.Issue.SearchWithContext(ctx, jql, options)
+	if err != nil {
+		if response != nil && response.StatusCode == http.StatusNotFound {
+			return nil, response, NotFoundError{err}
```

Comment:
I don't see any reason why `HandleJiraError` can't handle all those checks.
Do you think it makes sense to have a separate PR for this since I would need to change all the other methods where applicable(which might be out of scope), or is it fine to add it to this PR?

Reply:
Yep! I see this is a pattern that came before your PR - let's have that as a follow-up.

Reply:
Great, thanks! We'll create a follow-up PR.